include/seastar/core/sharded.hh

```diff
@@ -338,6 +340,42 @@ public:
       }
     }
 
+    /// Invoke a callable on a range of instances of `Service`.
+    ///
+    /// \param begin shard id that begins the range (inclusive)
+    /// \param end shard id that ends the range (exclusive)  
+    /// \param options the options to forward to the \ref smp::submit_to()
+    ///         called behind the scenes.
+    /// \param func a callable with signature `Value (Service&, Args...)` or
+    ///        `future<Value> (Service&, Args...)` (for some `Value` type), or a pointer
+    ///        to a member function of Service
+    /// \param args parameters to the callable; will be copied or moved. To pass by reference,
+    ///              use std::ref().
+    /// \return Future that becomes ready once all calls have completed
+    template <typename Func, typename... Args>
+    requires std::invocable<Func, Service&, Args...>
+        && std::is_same_v<futurize_t<std::invoke_result_t<Func, Service&, internal::sharded_unwrap_t<Args>...>>, future<>>
+    future<>
+    invoke_on(unsigned begin, unsigned end, smp_submit_to_options options, Func func, Args... args) noexcept;
```

Comment:
Better: pass a std::ranges::range of shards. Then any subset of the shards can be sent. If you want a contiguous range, use std::views::iota.